runners/twister2/src/main/java/org/apache/beam/runners/twister2/Twister2Runner.java

```diff
@@ -145,7 +151,16 @@ public PipelineResult runTest(Pipeline pipeline) {
     Twister2PipelineExecutionEnvironment env = new Twister2PipelineExecutionEnvironment(options);
     LOG.info("Translating pipeline to Twister2 program.");
     pipeline.replaceAll(getDefaultOverrides());
+    // TODO(BEAM-10670): Use SDF read as default when we address performance issue.
+    if (!ExperimentalOptions.hasExperiment(pipeline.getOptions(), "use_sdf_read")) {
+      // Populate experiments directly to have Kafka use legacy read.
+      ExperimentalOptions.addExperiment(
+          pipeline.getOptions().as(ExperimentalOptions.class), "beam_fn_api_use_deprecated_read");
```

Comment:
Looks like only Twister2 runner populates `use_deprecated_read`. Any reason?

Reply:
I did change it in https://github.com/apache/beam/pull/14849/files but it seems like this portion is not removed during cherry-pick.

Reply:
ok there are different portions: `run()` and `runTest`... I have to create another PR to fixup. Sorry for the inconvenience